### PR TITLE
macros: refactor histogram macros

### LINKED ISSUE #55: simplify histogram register macro
- use Vec<f64> or &[f64] for buckets in register histogram macro
- add bucket arg for register histogram vec macro

### PR DESCRIPTION
**Pull request by overvenus**
Macro API changes:

* `histogram_opts!`
  * Remove: 
    - `histogram_opts!( $ NAME : expr , $ HELP : expr , [ $ ( $ BUCKETS : expr ) , * ] )`
    - `histogram_opts!( $ NAME : expr , $ HELP : expr $ ( , $ CONST_LABELS : expr ) * )`
    - `histogram_opts!( $ NAME : expr , $ HELP : expr , $ CONST_LABELS : expr , [ $ ( $ BUCKETS : expr ) , + ] )`
  * Add:
    - `histogram_opts!( $ NAME : expr , $ HELP : expr )`
    - `histogram_opts!( $ NAME : expr , $ HELP : expr , $ BUCKETS : expr )`

* `register_histogram!`
  * Remove: 
    - `register_histogram!( $ NAME : expr , $ HELP : expr , $ CONST_LABELS : expr )`
    - `register_histogram!( $ NAME : expr , $ HELP : expr , $ CONST_LABELS : expr , [ $ ( $ BUCKETS : expr ) , + ] )`
  * Add:
    - `register_histogram!( $ NAME : expr , $ HELP : expr , $ BUCKETS : expr )`

* `register_histogram_vec!`
  * Remove: 
    - `register_histogram_vec!( $ NAME : expr , $ HELP : expr , $ CONST_LABELS : expr , $ LABELS_NAMES : expr )`
  * Add:
    - `register_histogram_vec!( $ NAME : expr , $ HELP : expr , $ LABELS_NAMES : expr , $ BUCKETS : expr)`

In a nutshell, `CONST_LABELS` are replaced with `BUCKETS`. 

Close #55 

PTAL @siddontang

### CODE REVIEW

**Comment by siddontang** [comment]
can we add CONST_LABELS to the end? can we put BUCKETS before the LABEL_NAMES?

**Comment by overvenus** [comment]
Then we have to assign buckets in order to add const_labels.

**Comment by siddontang** [comment]
Counter and Gauge have const in their register macro, but histogram has not, this is very strange. 
We should keep the style united.

I prefer removing const_labels in the register macro, because they are not commonly used. If user really want to use it, he can pass the Opt instead.

**Comment by siddontang** [comment]
Ping @overvenus

**Comment by overvenus** [comment]
Closed via #91
